### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by jonathanmunamire**
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+    <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
+  <% end %>
```
Comment (BenMukebo) [inline app/views/devise/registrations/edit.html.erb:13]: - [ ] Good job 👏🏻 however the views in devise for the registration and login pages should match the provided wireframes, kindly make sure to implement this so your project is completed. 💯 👍🏻 

|   |   |
|-|-|
|image|image|

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @jonathanmunamire, @TechnoAndy,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- The application allows for registering a new user. ✔️ 
- A user can log in with a combination of email and password✔️ 
- The application asks for confirmation of email. ✔️
- A user can reset the password in the application.🥇 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] I also recommend that you delete this 👇🏻 in your License section as these are instructions on how to complete the License section.

image

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/edit.html.erb (hunk 1 above)
Comment (BenMukebo) [reply app/views/devise/registrations/edit.html.erb:13]: ## Note: Don't merge this PR I made a mistake approving it on my dashboard, they will change that soon. In the meantime, you can implement the requested changes 👍🏻

**Comment by BenMukebo** [comment]
Hello @jonathanmunamire, @TechnoAndy, Please, let me know when you're done with the requested changes !

**Comment by jonathanmunamire** [comment]
Hello @BenMukebo , We're already done with the requested changes

**Comment by BenMukebo** [comment]
###  Hello @jonathanmunamire, @TechnoAndy 

- [ ] Let's mak sure the header is center on the center of your page before this project to be approved



- How about this one 👇🏻 

> 
> - [OPTIONAL] I also recommend that you delete this 👇🏻 in your License section as these are instructions on how to complete the License section.
> 
> image
>

**Comment by jonathanmunamire** [comment]
Hello @BenMukebo, the requested changes have already been implemented

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @jonathanmunamire, @TechnoAndy,**

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- The application allows for registering a new user. ✔️ 
- A user can log in with a combination of email and password✔️ 
- The application asks for confirmation of email. ✔️
- Your Design looks awesome, keep rocking :rocket: 
 👍  Good job!


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**